### PR TITLE
Optimized Courses Problems Endpoint

### PR DESCRIPTION
**Pull request by clintonb**
Using raw SQL to retrieve and aggregate the data. This significantly decreases response time. Also added a created field so that users know when the data was last updated.

@rocha @dsjen @jbau @mulby @brianhw @dylanrhodes

### CODE REVIEW

File: analytics_data_api/v0/views/courses.py [hunk 1]
```diff
@@ -627,36 +629,31 @@ class ProblemsListView(BaseCourseView):
...
+GROUP BY module_id;
+        """
+        with connections[settings.ANALYTICS_DATABASE].cursor() as cursor:
+            cursor.execute(sql, [self.course_id])
+            rows = dictfetchall(cursor)
 
-        data = []
+        for row in rows:
+            # Convert the comma-separated list into an array of strings.
+            row['part_ids'] = row['part_ids'].split(',')
```
Comment (dsjen) [inline analytics_data_api/v0/views/courses.py]: The previous code had `part_ids = set()    # Use a set to remove duplicate values.`.  It seems this change may introduce duplicates.

Comment (clintonb) [inline analytics_data_api/v0/views/courses.py]: The SQL query  has a distinct modifier on the `part_ids` value returned.

Comment (dsjen) [inline analytics_data_api/v0/views/courses.py]: Excellent!

**Comment by dsjen** [comment]
Looks like the build is failing.  Otherwise, looks good to me!

**Pull request merged**